File: SO3/tests/testSO3vec.cpp
#include "GElib_base.cpp"
//#include "GElibSession.hpp"
#include "SO3vec.hpp"
#include "SO3functions.hpp"

using namespace cnine;
using namespace GElib;

int main(int argc, char** argv){
  //GElibSession session;
  cout<<endl;

  SO3vec<float> u(ttype=SO3type({{2,1},{3,1}}),filltype=4);
  SO3vec<float> v(ttype=SO3type({{2,1},{3,1}}),filltype=4);
  print(v);

  auto w=CGproduct(u,v);
  print(w);

  auto z=DiagCGproduct(u,v);
  print(z);

}

File: python/tests/test_SO3vec.py
import torch
import gelib as g

A=g.SO3vec.randn(1,{1:2,2:2})
Ad=g.SO3vec.randn(1,{1:2,2:2})
B=g.SO3vec.randn(3,{2:1,3:1})
print(A.__repr__())
print(A)
print(B)

R0=g.SO3element.random()
A1=A.apply(R0)
print(A1)

C=g.CGproduct(A,A)
print(C)

E=g.CGproduct(A,B)
print(E)

D=g.DiagCGproduct(A,Ad,2)
print(D)

